lib/components/map/connected-transitive-overlay.js

```diff
@@ -1,44 +1,71 @@
+import isEqual from 'lodash.isequal'
 import coreUtils from '@opentripplanner/core-utils'
 import TransitiveCanvasOverlay from '@opentripplanner/transitive-overlay'
+import React, { Component } from 'react'
 import { connect } from 'react-redux'
 
-import { getActiveSearch, getActiveItinerary, getActiveItineraries } from '../../util/state'
+import { ComponentContext } from '../../util/contexts'
+import { getActiveItinerary, getActiveItineraries, getActiveSearch } from '../../util/state'
+
+/**
+ * Wrapper for TransitiveCanvasOverlay that passes getTransitiveRouteLabel defined in ComponentContext.
+ */
+class TransitiveCanvasOverlayWithContext extends Component {
+  static contextType = ComponentContext
+
+  shouldComponentUpdate (nextProps) {
+    return (
+      !isEqual(nextProps.activeSearchVisibleItinerary, this.props.activeSearchVisibleItinerary) ||
+      !isEqual(nextProps.activeSearchResponseOtp, this.props.activeSearchResponseOtp)
```

Comment:
This could end up being quite an expensive operation because of the deep comparisons done to look at each of these items. It'd be much better to have a simple equality check or at least not check so many items.